presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -330,6 +330,7 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                 stats,
                 ImmutableList.of(new PartialAggregationPushDown(metadata.getFunctionRegistry())),
                 ImmutableSet.of(new PushPartialAggregationThroughJoin(), new PushPartialAggregationThroughExchange(metadata.getFunctionRegistry()))));
+        builder.add(new PruneUnreferencedOutputs());
```

Comment:
Is that needed once you restrict outputs? If so maybe there is a migrated version you can use? See com/facebook/presto/sql/planner/PlanOptimizers.java:144 and com.facebook.presto.sql.planner.iterative.rule.PruneJoinColumns